src/models/geotimeline.py

```diff
@@ -1,6 +1,522 @@
+            time1 (TimeStamp): Time stamp of the first graph (src_graph)
+            time2 (TimeStamp): Time stamp of the second graph (trg_graph)
+
+        Raises:
+            NotCachedError: If the combination (time1, time2) or its inverse
+                (time2, time1) have not been cached yet.
+
+        Returns:
+            NodeMap: The NodeMap to identify nodes from `self[time1]` with `self[time2]`
+        """
+        if (time1, time2) in self._node_map_cache.keys():
+            return self._node_map_cache[(time1, time2)]
+        elif (time2, time1) in self._node_map_cache.keys():
+            map_from_inverse = self._node_map_cache[(time2, time1)].invert()
+            self._node_map_cache[(time1, time2)] = map_from_inverse
+            return map_from_inverse
+        else:
+            raise NotCachedError
+
+    def _empty_node_map_cache(self) -> None:
+        """ Empties the node map cache."""
+        self._node_map_cache = dict()
+
+    def timestack(self, use_cached: bool = True) -> List[NodeMap]:
+        """
+        Performs node identification between adjacent time-slices in the graph.
+
+        Args:
+            use_cached (bool, optional): If True, reuses prior node-identification
+                computations. Defaults to True.
+
+        Returns:
+            List[NodeMap]: An ordered list of the of the node maps between each two
+                adjacent time-slices in the GeoGraphTimeline
+        """
+        node_maps = []
+        for time1, time2 in zip(self.times, self.times[1:]):
+            node_maps.append(self.identify_graphs(time1, time2, use_cached))
+        return node_maps
+
+    def timediff(self, use_cached: bool = True):
+        raise NotImplementedError
+
+    def node_diff_cache(self, time1: TimeStamp, time2: TimeStamp):
         raise NotImplementedError
+
+    def get_metric(self, name: str, class_value: Optional[int] = None) -> xr.DataArray:
+        """
+        Return the time-series for the given metric.
+
+        For class-level metrics pass a class_value argument. For landscape/component
+        level metrics omit the class_value argument.
+
+        Args:
+            name (str): Name of the metric to compute
+            class_value (Optional[int], optional): Provide a class value if you wish
+            to calculate a class-level metric. Leave as None for calculating
+            landscape/habitat level metrics. Defaults to None.
+
+        Returns:
+            xr.DataArray: A DataArray containing the metric time series for the graphs
+                in the given GeoGraphTimeline
+        """
+
+        # Calculate metrics
+        metrics = [
+            graph.get_metric(name=name, class_value=class_value) for graph in self
+        ]
+
+        # Set up metadata
+        attrs = {"description": metrics[0].description, "unit": metrics[0].unit}
+        if class_value:
+            attrs["class_label"] = class_value
+
+        metric_timeseries = xr.DataArray(
```

Comment:
I'm curious - what makes the `DataArray` more convenient here?

Reply:
The multidimensionality and the fact that we can add the nice description that you wrote for the metrics as well as units to it 😉